services/QuillLMS/client/app/bundles/Diagnostic/components/questions/response.tsx

```diff
@@ -0,0 +1,446 @@
+      return;
+    }
+    if (response.parentID || response.parent_id) {
+      const responseParent = parent;
+      if (!responseParent) {
+        getParentResponse(response.parentID || response.parent_id)
+        parentDetails = [
+          (<p>Loading...</p>),
+          (<br />)
+        ]
+      } else {
+        const diffText = applyDiff(responseParent.text, response.text);
+        parentDetails = [
+          (<span><strong>Parent Text:</strong> {responseParent.text}</span>),
+          (<br />),
+          (<span><strong>Parent Feedback:</strong> {responseParent.feedback}</span>),
+          (<br />),
+          (<span><strong>Differences:</strong> {diffText}</span>),
+          (<br />),
+          (<br />)
+        ];
+      }
+    }
+
+    if (isEditing) {
+      content =
+        (<div className="content">
+          {parentDetails}
+          <label className="label">Feedback</label>
+          <TextEditor
+            boilerplate={selectedBoilerplate}
+            ContentState={ContentState}
+            EditorState={EditorState}
+            handleTextChange={handleFeedbackChange}
+            text={feedback || ''}
+          />
+
+          <br />
+
+          <div className="box">
+            <label className="label">Concept Results</label>
+            {renderConceptResults('Editing')}
+          </div>
+
+          <p className="control">
+            <label className="checkbox">
+              <input defaultChecked={response.optimal} ref={newResponseOptimal} type="checkbox" />
+              Optimal?
+            </label>
+          </p>
+        </div>);
+    } else {
+      content =
+        (<div className="content">
+          {parentDetails}
+          <strong>Feedback:</strong> <br />
+          <div dangerouslySetInnerHTML={{ __html: response.feedback, }} />
+          <br />
+          <label className="label">Concept Results</label>
+          <ul>
+            {renderConceptResults('Viewing')}
+          </ul>
+          {authorDetails}
+          {childDetails}
+          {pathwayDetails}
+        </div>);
+    }
+
+    return (
+      <div className="card-content">
+        {content}
+      </div>
+    );
+  };
+
+  function renderResponseFooter(isEditing, response) {
+    if (!readOnly || !expanded) {
+      return;
+    }
+    let buttons;
+
+    if (isEditing) {
+      buttons = [
+        (<a className="card-footer-item" key="cancel" onClick={() => cancelResponseEdit(response.key)} >Cancel</a>),
+        (<a className="card-footer-item" key="unmatch" onClick={() => unmatchResponse(response.key)} >Unmatch</a>),
+        (<a className="card-footer-item" key="update" onClick={() => updateResponse(response.key)} >Update</a>)
+      ];
+    } else {
+      buttons = [
+        (<a className="card-footer-item" key="edit" onClick={() => editResponse(response.key)} >Edit</a>),
+        (<a className="card-footer-item" key="delete" onClick={() => handleDeleteResponse(response.key)} >Delete</a>)
+      ];
```

Comment:
You can take or leave this, but I think in all these onClick functions, because the response is in the state, we could extract the `response.key` inside of the function that's getting called and then we don't have to pass an argument to the onClick handler and can rewrite them as `onClick={cancelResponseEdit}`, which obeys our linter rules and is easier to read.